### PR TITLE
Add PhpdocNoUselessInheritdocFixer

### PR DESCRIPTION
**Pull request by SpacePossum**
Early opening, need to fix the priority tests.
For ref.: https://github.com/twigphp/Twig/pull/2254/files

### CODE REVIEW

**Comment by keradus** [comment]
what if documented `inheritdoc` method comes from trait, not parent class ? could it be a case ?

**Comment by SpacePossum** [comment]
>what if documented inheritdoc method comes from trait, not parent class ? could it be a case ?

good point, I'll add a test for candidate if a class uses a trait

**Comment by localheinz** [comment]
Nice!

**Comment by SpacePossum** [comment]
all points addressed

File: src/Fixer/Phpdoc/PhpdocNoUselessInheritdocFixer.php [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        // min. offset 4 as minimal candidate is @: <?php\n/** @inheritdoc */class min{}
+        for ($index = 1, $count = count($tokens) - 4; $index < $count; ++$index) {
+            if (!$tokens[$index]->isGivenKind(array(T_CLASS, T_INTERFACE))) {
+                continue;
+            }
+
+            $index = $this->fixClassy($tokens, $index);
```
Comment (keradus) [inline src/Fixer/Phpdoc/PhpdocNoUselessInheritdocFixer.php]: remove negation from condition, move this inside the condition, 2 lines less and no negation ;)

File: src/Fixer/Phpdoc/PhpdocNoUselessInheritdocFixer.php [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Classy that does not inherit must not have inheritdoc tags.',
+            array(
+                new CodeSample("<?php\n/** {@inheritdoc} */\nclass MyClass\n{\n}"),
+            )
```
Comment (keradus) [inline src/Fixer/Phpdoc/PhpdocNoUselessInheritdocFixer.php]: please add a case with a method to show that it also works for them

File: tests/FixerFactoryTest.php [hunk 3]
```diff
@@ -300,6 +300,11 @@ public function getFixersPriorityCases()
             array($fixers['protected_to_private'], $fixers['ordered_class_elements']), // tested also in: protected_to_private,ordered_class_elements.test
             array($fixers['phpdoc_add_missing_param_annotation'], $fixers['phpdoc_align']), // tested also in: phpdoc_add_missing_param_annotation,phpdoc_align.test
             array($fixers['phpdoc_no_alias_tag'], $fixers['phpdoc_add_missing_param_annotation']), // tested also in: phpdoc_no_alias_tag,phpdoc_add_missing_param_annotation.test
+            array($fixers['phpdoc_no_useless_inheritdoc'], $fixers['no_empty_phpdoc']), // tested also in: phpdoc_no_useless_inheritdoc,no_empty_phpdoc.test
+            array($fixers['phpdoc_no_useless_inheritdoc'], $fixers['phpdoc_inline_tag']), // tested also in: phpdoc_no_useless_inheritdoc,phpdoc_inline_tag.test
+            array($fixers['phpdoc_to_comment'], $fixers['phpdoc_no_useless_inheritdoc']), // tested also in: phpdoc_to_comment,phpdoc_no_useless_inheritdoc.test
+
+                        // FIXME if skip priority, do it anyway to see if the claim correct
```
Comment (keradus) [inline tests/FixerFactoryTest.php]: ?

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request closed without merging**